Framework/Geometry/test/CSGObjectTest.h

```diff
@@ -1143,8 +1143,8 @@ class CSGObjectTest : public CxxTest::TestSuite {
 
     origin = Track(V3D{0., 0., 0.}, BEAM_DIRECTION);
     nsegments = cylinder->interceptSurface(origin);
-    TS_ASSERT_EQUALS(nsegments, 0);
-    TS_ASSERT_EQUALS(origin.totalDistInsideObject(), 0.0);
+    TS_ASSERT_EQUALS(nsegments, 1);
+    TS_ASSERT_EQUALS(origin.totalDistInsideObject(), 0.0285528019);
```

Comment:
I've just drawn myself a picture of the cylinder and the track for this final test and it looks like nsegments should equal zero (and totalDistInsideObject should equal 0.0). I wonder if the missing check on TrackDirection::INVALID in CSGObject.cpp is the problem here